org.adempiere.base/src/org/compiere/print/ReportEngine.java

```diff
@@ -1203,7 +1178,91 @@ else if (obj instanceof PrintData)
 		return true;
 	}	//	createHTML
 
-
+	/**
+	 * Get record identifier string
+	 * @param mTable
+	 * @param tableName
+	 * @param recordID
+	 * @return String identifier
+	 */
+	private String getIdentifier(MTable mTable, String tableName, int recordID) {
+		ArrayList<MColumn> list = new ArrayList<MColumn>();
+		MClient tenant = MClient.get(getCtx(), Env.getAD_Client_ID(getCtx()));
+		// check translation table
+		MTable mTableTrl = MTable.get(getCtx(), tableName+"_Trl");
+		String tableNameTrl = "";
+		// get languages
+		String tenantLang = tenant.getAD_Language();
+		String reportLang = new MLanguage(getCtx(), getLanguageID(), null).getAD_Language();
+		
+		// use Trl table
+		boolean isTrl = !tenantLang.equalsIgnoreCase(reportLang);
```

Comment:
Please check that other part of report engine is using the Env.isBaseLanguage (String AD_Language, String tableName) call instead.

Reply:
Hi @hengsin, I have not found any uses of Env.isBaseLanguage (String AD_Language, String tableName) in report engine.

I've checked the method, and it does not give the same result as the committed code. The point would be here, to compare the report language and the Tenant's language, and search in the Trl table only if they are not equal. E.g. if Tenant's language is spanish, and report language is also spanish, than get the value from the base table, do not search from the Trl table.

However, with Env.isBaseLanguage it does not work, since the base language is still en_US even if I change Tenant language to es_CO.

Wdyt? Could it stay this way?

Reply:
It is use in DataEngine

Reply:
Yes but Env.isBaseLanguage does not use AD_Client.AD_Language to compare. Is it not better to use AD_Client.AD_Language here?

Reply:
I didn't mean which is better, I'm just pointing out the inconsistency here. We can't have 2 different approach in the report engine.